### PR TITLE
Add warning for LXD

### PR DESCRIPTION
**Pull request by evilnick**
This adds:

 - a section in the troubleshooting page, detailing the problem and a workaround
 - warning at the top of 1.20 upgrade page
 -  warning in the upgrade notes

### CODE REVIEW

**Comment by evilnick** [comment]
@chris-sanders this is the preview of the main change - https://deploy-preview-523--cdk-docs-next.netlify.app/kubernetes/docs/troubleshooting#lxd , the other parts are mainly warnings pointing to this.
I believe I understood the issue and resolution, but please check ;)

File: pages/k8s/upgrading.md [hunk 1]
```diff
@@ -19,9 +19,9 @@ toc: False
   <p markdown="1" class="p-notification__response">
     <span class="p-notification__status">Note:</span>
 This page describes the general upgrade process. It is important to follow the specific upgrade pages for each release, as these may include additional steps and workarounds for safely upgrading. <br><br>
-<a class='p-button--brand' href='/kubernetes/docs/1.19/upgrading'>Upgrade to 1.19 </a>
-<a class='p-button--brand' href='/kubernetes/docs/1.18/upgrading'>Upgrade to 1.18 </a>
-<a class='p-button--brand' href='/kubernetes/docs/1.17/upgrading'>Upgrade to 1.17 </a>
+<a class='p-button--brand' href='/kubernetes/docs/1.19/upgrading'>Upgrade to 1.20 </a>
```
Comment (kwmonroe) [inline pages/k8s/upgrading.md:22]: looks like the hrefs are pointing to `n-1` for this and the next 2 links.

**Review by kwmonroe — CHANGES_REQUESTED** [review]
LXD bits look good, but it seems this introduces incorrect `hrefs` in the `upgrading.md` page.

File: pages/k8s/upgrading.md (hunk 1 above)
Comment (evilnick) [reply pages/k8s/upgrading.md:22]: thx

Comment (evilnick) [reply pages/k8s/upgrading.md]: @kwmonroe fixed

**Review by johnsca — CHANGES_REQUESTED** [review]
This repeatedly mentions that it affects "clouds running on LXD" or "the localhost cloud" but my understanding of the original issue is actually that it was discovered because of placement of  units into LXD containers on other machines / instances in the same model.

If my understanding is correct, I think the current wording will lead people to ignore it if they're running on MAAS or other clouds, even if they are using LXD placement.

**Comment by evilnick** [comment]
I changed the wording to include other lxd-based deployments

**Comment by chris-sanders** [comment]
lgtm

**Pull request merged**